Fix bug in ImageBuilder resources tagging

jdeamicis opened the pull request:
### Description of changes
* Set the build tags also in the resources created by ImageBuilder.

### Tests
* Modified imagebuilder_stack unit tests accordingly
* Added assertion in `test_build_image()` and ran the integration test to verify that the new assertion works as expected.

### References
Fixes known issue: https://github.com/aws/aws-parallelcluster/issues/5089

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] ~Check if documentation is impacted by this change.~

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

jdeamicis committed:
Fix bug in ImageBuilder resources tagging

Set the build tags also in the resources created by ImageBuilder.

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>

lukeseawalker commented on cli/tests/pcluster/templates/test_imagebuilder_stack.py:
```diff
@@ -2577,6 +2577,9 @@ def test_imagebuilder_build_tags(mocker, resource, response, expected_imagebuild
             or resource_name == "BuildNotificationTopic"
         ):
             assert_that(resource.get("Properties").get("Tags")).is_equal_to(expected_role_tags)
+        elif resource_name == "InfrastructureConfiguration":
```
instead of adding this elif, you can add an if separated, so that the check on "Tags" isn't repeated

jdeamicis replied:
Addressing in the next push.

jdeamicis committed:
Add check for build tags to test_build_image test

Check that resources (instances) created by ImageBuilder are tagged
with the build tags provided in the ParallelCluster image configuration
file.

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>

lukeseawalker commented on tests/integration-tests/tests/createami/test_createami.py:
```diff
@@ -141,6 +142,7 @@ def test_build_image(
     )
 
     _test_build_image_success(image)
+    _test_build_instances_tags(image, image.config["Build"]["Tags"], region)
```
I think this will be a noop, since the previous `_test_build_image_success` waits for BUILD_COMPLETE, there are no more instances at this execution point

lukeseawalker replied:
unless the describe_instances is also returning the just terminated instances

jdeamicis replied:
`describe_instances` *may* return recently terminated instances (see [API documentation](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstances.html)).

Currently in the test we are already running a very similar check on IMDSv2, and it relies on a describe_instances of the very same ImageBuilder instances that are terminated at the end of the build-image process. That test currently works, and also the new check I just introduced works for the moment.

enrico-usai committed:
Merge branch 'develop' into fix/devel/imagebuilder_resource_tags
